src/olympia/scanners/models.py

```diff
@@ -177,6 +231,51 @@ def clean_yara(self):
             )
 
 
+class ScannerRule(AbstractScannerRule):
+    class Meta(AbstractScannerRule.Meta):
+        db_table = 'scanners_rules'
+
+
+class ScannerResult(AbstractScannerResult):
+    upload = models.ForeignKey(
+        FileUpload,
+        related_name="%(class)ss",  # scannerresults
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+
+    class Meta(AbstractScannerResult.Meta):
+        db_table = 'scanners_results'
+        constraints = [
+            models.UniqueConstraint(
+                fields=('upload', 'scanner', 'version'),
+                name='scanners_results_upload_id_scanner_'
+                'version_id_ad9eb8a6_uniq',
+            )
+        ]
+
+
 class ScannerMatch(ModelBase):
     result = models.ForeignKey(ScannerResult, on_delete=models.CASCADE)
     rule = models.ForeignKey(ScannerRule, on_delete=models.CASCADE)
+
+
+class ScannerQueryRule(AbstractScannerRule):
+    class Meta(AbstractScannerRule.Meta):
+        db_table = 'scanners_query_rules'
+
+
+class ScannerQueryResult(AbstractScannerResult):
+    # Has to be overridden, because the parent refers to ScannerMatch.
+    matched_rules = models.ManyToManyField(
+        'ScannerQueryRule', through='ScannerQueryMatch'
+    )
+
+    class Meta(AbstractScannerResult.Meta):
+        db_table = 'scanners_query_results'
+        # FIXME indexes, unique constraints ?
```

Comment:
Is this comment for later?

Reply:
Yeah. I'm probably going to add more things, and I didn't want to do multiple migrations fiddling with the indexes just now, so I left it for later.